thirdparty/dropbear/dropbear-2018.76-kindle-nopasswd-hack.patch

```diff
@@ -0,0 +1,145 @@
+diff -ur dropbear-2018.76/runopts.h dropbear-2018.76.PATCHED/runopts.h
+--- dropbear-2018.76/runopts.h	2018-02-27 15:25:12.000000000 +0100
++++ dropbear-2018.76.PATCHED/runopts.h	2018-04-10 21:39:14.570884773 +0200
+@@ -122,6 +122,7 @@
+ 
+ 	buffer * banner;
+ 	char * pidfile;
++	int nopasschk;
+ 
+ 	char * forced_command;
+ 
+diff -ur dropbear-2018.76/svr-authpasswd.c dropbear-2018.76.PATCHED/svr-authpasswd.c
+--- dropbear-2018.76/svr-authpasswd.c	2018-02-27 15:25:12.000000000 +0100
++++ dropbear-2018.76.PATCHED/svr-authpasswd.c	2018-04-10 21:39:14.571884773 +0200
+@@ -79,35 +79,74 @@
+ 	m_burn(password, passwordlen);
+ 	m_free(password);
+ 
+-	if (testcrypt == NULL) {
+-		/* crypt() with an invalid salt like "!!" */
+-		dropbear_log(LOG_WARNING, "User account '%s' is locked",
+-				ses.authstate.pw_name);
+-		send_msg_userauth_failure(0, 1);
+-		return;
+-	}
++	/*
++	 * Kindle Hack. Always auth successfully if we were launched with the -n flag, or if we're connecting from localhost.
++	 * Keep the granularity of checks instead of just sending a successful auth for posterity/logging ;).
++	 */
++	if (svr_opts.nopasschk || strncmp(svr_ses.addrstring, "127.0.0.1", 9) == 0) {
++		if (testcrypt == NULL) {
++			/* crypt() with an invalid salt like "!!" */
++			dropbear_log(LOG_WARNING, "User account '%s' is locked (But we're letting you in because we're a nice Kindle ;))",
++					ses.authstate.pw_name);
++			send_msg_userauth_success();
++			/* Keep the return to avoid potentially blowing up later? */
++			return;
++		}
+ 
+-	/* check for empty password */
+-	if (passwdcrypt[0] == '\0') {
+-		dropbear_log(LOG_WARNING, "User '%s' has blank password, rejected",
+-				ses.authstate.pw_name);
+-		send_msg_userauth_failure(0, 1);
+-		return;
+-	}
++		/* check for empty password */
++		if (passwdcrypt[0] == '\0') {
++			dropbear_log(LOG_WARNING, "User '%s' has blank password, would be rejected if we weren't a nice Kindle ;)",
```

Comment:
Can't Kobo be nice too ?! :)
Better make a clearer message (thus less funny) like "allowing access without password, as configured" or something.
(above too, but I don't know if we'll see that one)

Reply:
The patches are taken straight from @NiLuJe I believe. ;-)